src/models/device-type.ts

```diff
@@ -146,20 +150,18 @@ const getDeviceTypeModel = function (deps: InjectedDependenciesParam) {
 		 * 	console.log('resolved alias:', deviceType);
 		 * });
 		 */
-		async get(
+		async get<T extends ODataOptionsWithoutCount<DeviceType['Read']>>(
 			idOrSlug: number | string,
-			options?: PineOptions<DeviceType>,
-		): Promise<DeviceType> {
-			options ??= {};
-
+			options?: T,
+		) {
 			if (idOrSlug == null) {
 				throw new errors.BalenaInvalidDeviceType(idOrSlug);
 			}
 
-			let deviceType: DeviceType | undefined;
+			let deviceType: OptionsToResponse<DeviceType['Read'], T, typeof idOrSlug>;
```

Comment:
I don't know why, but it seems that this needs the `| undefined`

![Image](https://github.com/user-attachments/assets/c6494ac8-65f1-47dd-9aaf-85da018afe00)